docs/tutorial/category_theory/calculating_colimits_in_Top.lean

```diff
@@ -61,28 +72,32 @@ section Gluing
 
 -- Here's two copies of the real line glued together at a point.
 def f : pt ⟶ R := { val := λ _, (0 : ℝ), property := continuous_const }
+/-- Two copies of the real line glued together at 0. -/
 def X : Top := pushout f f
 
 -- To define a map out of it, we define maps out of each copy of the line,
 -- and check the maps agree at 0.
 def g : X ⟶ R :=
-pushout.desc f f (𝟙 _) (𝟙 _) rfl
+pushout.desc (𝟙 _) (𝟙 _) rfl
 
 end Gluing
 
 universes v u w
 
 section Products
 
-def d : discrete ℕ ⥤ Top := functor.of_function (λ n : ℕ, R)
-
-def Y : Top := limit d
-
-def w : cone d := fan.mk (λ (n : ℕ), ⟨λ (_ : pt), (n : ℝ), continuous_const⟩)
+/-- The countably infinite product of copies of `ℝ`. -/
+def Y : Top := ∏ (λ n : ℕ, R)
 
+/-- We define a point of this infinite product by specifying its coordinates. -/
 def q : pt ⟶ Y :=
-limit.lift d w
+pi.lift (λ (n : ℕ), ⟨λ (_ : pt), (n : ℝ), continuous_const⟩)
+
+-- "Looking under the hood", we see that `q` is a `subtype`, whose `val` is a function `unit → Y.α`.
+#check q.val
```

Comment:
Please comment out this `#check`.

Reply:
Notice this is in a `docs/tutorial/` file. I think `#check` and `#print` are actually good useful in such files.

Reply:
I thought we didn't want mathlib to have any output. That's why I suggest commenting it. People reading the file can easily remove the comment. @digama0 @robertylewis What's the best thing to do here? (I'm not sure about what we want.)

Reply:
But, it's not really "in mathlib". These files are not compile when you build or test mathlib.

Reply:
I would venture the opinion that if all the tutorials and tests are checked routinely and that each one `#print`s freely, the output is going to be pretty noisy and unhelpful.

Reply:
Tests which get compiled automatically I somewhat understand, but personally I don't see how a doc file like this is any different to making a `blah.md` with this entire code pasted in it.

Reply:
Certainly if `/doc/` is going to be routinely tested, we should suppress output. At the moment it isn't, however, so I thought it was safe to leave instructive `#check` in. The best solution, of course, is to comment out the #check, and include the expected output as another comment, which I'll do now.

Reply:
Ok, I agree. Let's leave it like this for now. We do run the risk that the next refactor will forget about this file, and so it might get out of sync. I was under the assumption that these files were also checked by Travis etc...